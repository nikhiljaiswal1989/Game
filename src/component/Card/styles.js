import { StyleSheet } from 'react-native';
import { color, dimen } from '../../constants';
const styles = StyleSheet.create({
  cardContainer: {
    height: dimen.cardHeight,
    width: dimen.cardWidth,
    borderWidth: 4,
    borderColor: 'transparent',
    padding: 4,
  },
  cardNotOpen: {
    flex: 1,
    backgroundColor: color.primaryColor,
    padding: 8,
    borderRadius: 8,
    borderWidth: 4,
    borderColor: '#fff',
    justifyContent: 'center',
    alignItems: 'center',
  },
  cardOpen: {
    flex: 1,
    backgroundColor: '#fff',
    padding: 8,
    borderRadius: 8,
    borderWidth: 4,
    borderColor: '#fff',
    justifyContent: 'center',
    alignItems: 'center',
  },
  openNumber: {
    fontSize: 24,
    color: '#000',
  },
  closeNumber: {
    fontSize: 32,
    color: '#fff',
    fontWeight: 'bold',
  },
  flipCard: {
    alignItems: 'center',
    justifyContent: 'center',
    backfaceVisibility: 'hidden',
  },
  flipCardBack: {
    position: 'absolute',
    top: 0,
  },
});

export default styles;
