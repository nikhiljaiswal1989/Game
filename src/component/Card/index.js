import React, { useState } from 'react';
import { View, Text, TouchableWithoutFeedback, Animated } from 'react-native';
import styles from './styles';
const Card = ({ item, onClick, testID }) => {
  const isOpen = item.isOpen || item.isMatchFound;
  const [animatedValue] = useState(new Animated.Value(0));

  isOpen
    ? Animated.spring(animatedValue, {
        toValue: 0,
        friction: 8,
        tension: 10,
        useNativeDriver: true,
      }).start()
    : Animated.spring(animatedValue, {
        toValue: 180,
        friction: 8,
        tension: 10,
        useNativeDriver: true,
      }).start();

  animatedValue.addListener(({ value }) => {
    value = value;
  });
  let frontInterpolate = animatedValue.interpolate({
    inputRange: [0, 180],
    outputRange: ['0deg', '180deg'],
  });
  let backInterpolate = animatedValue.interpolate({
    inputRange: [0, 180],
    outputRange: ['180deg', '360deg'],
  });
  let frontOpacity = animatedValue.interpolate({
    inputRange: [89, 90],
    outputRange: [1, 0],
  });
  let backOpacity = animatedValue.interpolate({
    inputRange: [89, 90],
    outputRange: [0, 1],
  });

  const frontAnimatedStyle = {
    transform: [{ rotateY: frontInterpolate }],
  };
  const backAnimatedStyle = {
    transform: [{ rotateY: backInterpolate }],
  };

  return (
    <TouchableWithoutFeedback
      testID={testID}
      onPress={() => {
        !isOpen && onClick();
      }}>
      <View>
        <Animated.View
          style={[
            styles.flipCard,
            frontAnimatedStyle,
            { opacity: frontOpacity },
          ]}>
          <View style={styles.cardContainer}>
            <View style={styles.cardOpen}>
              <Text style={styles.openNumber}>{item.text}</Text>
            </View>
          </View>
        </Animated.View>
        <Animated.View
          style={[
            styles.flipCard,
            styles.flipCardBack,
            backAnimatedStyle,
            { opacity: backOpacity },
          ]}>
          <View style={styles.cardContainer}>
            <View style={styles.cardNotOpen}>
              <Text style={styles.closeNumber}>?</Text>
            </View>
          </View>
        </Animated.View>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default React.memo(Card);
