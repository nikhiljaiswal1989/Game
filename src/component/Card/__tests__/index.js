import React from 'react';
import { render } from '@testing-library/react-native';
import Card from '../index';
describe('DottedLine component', () => {
  it('renders', () => {
    const item = { isMatchFound: false, isOpen: true, text: 'text' };
    const tree = render(<Card item={item} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
