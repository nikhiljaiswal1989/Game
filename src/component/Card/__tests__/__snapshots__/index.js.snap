// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DottedLine component renders 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backfaceVisibility": "hidden",
        "justifyContent": "center",
        "opacity": 90,
        "transform": Array [
          Object {
            "rotateY": "0deg",
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "transparent",
          "borderWidth": 4,
          "height": 296,
          "padding": 4,
          "width": 250,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#fff",
            "borderRadius": 8,
            "borderWidth": 4,
            "flex": 1,
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 24,
            }
          }
        >
          text
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backfaceVisibility": "hidden",
        "justifyContent": "center",
        "opacity": -89,
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "rotateY": "180deg",
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "transparent",
          "borderWidth": 4,
          "height": 296,
          "padding": 4,
          "width": 250,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4AA2EE",
            "borderColor": "#fff",
            "borderRadius": 8,
            "borderWidth": 4,
            "flex": 1,
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 32,
              "fontWeight": "bold",
            }
          }
        >
          ?
        </Text>
      </View>
    </View>
  </View>
</View>
`;
